src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { Persona } from '../Persona/Persona';
+import { TableHeadLink } from '../TableHeadLink';
+
+type Props = {
+  filteredPeopleList: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  filteredPeopleList,
+}) => {
+  const [searchParams] = useSearchParams();
+  const sortColumn = searchParams.get('sort') || '';
+  const sortReverse = searchParams.get('order') || '';
+  const [
+    visiblePeopleList,
+    setVisiblePeopleList,
+  ] = useState<Person[]>(filteredPeopleList);
+
+  const tableHeadFilter = async () => {
+    let filteredPeople: Person[] = [...filteredPeopleList].sort((
+      personA: Person,
+      personB: Person,
+    ) => {
+      switch (sortColumn) {
+        case 'name':
+        case 'sex':
+          return personA[sortColumn].localeCompare(personB[sortColumn]);
+
+        case 'born':
+        case 'died':
+          return personA[sortColumn] - personB[sortColumn];
+
+        default:
+          return 0;
+      }
+    });
+
+    if (sortReverse === 'desc') {
+      filteredPeople = filteredPeople.reverse();
+    }
+
+    setVisiblePeopleList(filteredPeople);
+  };
+
+  useEffect(() => {
+    tableHeadFilter();
+  }, [filteredPeopleList, searchParams]);
```

Comment:
Better to wrap this function into useCallback and move these dependencies to that array. And here leave only function
```suggestion
    tableHeadFilter();
  }, [tableHeadFilter]);
```